frontend/src/components/script-editor.tsx

```diff
@@ -110,9 +116,50 @@ export default function ScriptEditor() {
         "scrollbarSlider.hoverBackground": "#258F4245",
       },
     });
+
     monaco.editor.remeasureFonts();
   };
 
+  const [currentDecorations, setCurrentDecorations] = useState<string[]>([]);
+
+  const [monaco, setMonaco] = useState<any>();
+
+  useEffect(() => {
+    if (monaco) {
+      const editorModel = monaco.editor.getModels()[0];
+      editorModel.deltaDecorations(currentDecorations, []);
+      const words = scriptPubKey.split(" ");
+
+      if (step >= 0 && step < words.length) {
+        const wordToHighlight = words[step];
+        const matches = editorModel.findMatches(wordToHighlight, false, false, false, null, false);
+
+        if (matches.length > 0) {
+          let matchIndexToUse = 0;
+
+          if (matches.length > 1) {
+            const occurrencesSoFar = words.slice(0, step + 1).filter(word => word === wordToHighlight).length;
```

Comment:
This works for now, but there will be script layouts that we will want to support later that will be a lot harder to do like this. Could you see if you can create a `range` value programmatically without using findMatches?
something like `let range = {start: words[step], end: words[step].len}` ( It wont work exactly like this, but I think it shows what I mean )

Reply:
I will look into this asap and get back to you